apps/dsb-message-broker/src/auth/dynamic.roles.guard.ts

```diff
@@ -14,23 +14,42 @@ export class DynamicRolesGuard implements CanActivate {
         const requiredRoleCategory = this.reflector.get<string>('role', context.getHandler());
         if (!requiredRoleCategory) return true;
 
+        const user = context.switchToHttp().getRequest().user;
+        const verifiedRoles = user.verifiedRoles?.map((role: any) => role.namespace) || [];
+        if (!verifiedRoles.length) return false;
+
         const organizations = this.configService.get('ORGANIZATIONS');
 
         const fqcn =
             context.switchToHttp().getRequest().body?.fqcn ??
             context.switchToHttp().getRequest().params?.fqcn ??
             context.switchToHttp().getRequest().query?.fqcn;
-        if (!fqcn) return false;
 
-        const { org, app } = extractFqcn(fqcn);
-        const requiredRoles = (
-            organizations
-                .find((_org: any) => _org.name === org)
-                ?.apps.find((_app: any) => _app.name === app)?.roles[requiredRoleCategory] || []
-        ).map((role: string) => `${role}.roles.${app}.apps.${org}.iam.ewc`);
+        let requiredRoles: string[] = [];
+        if (!fqcn) {
```

Comment:
didn't quite get what's going on here

Reply:
if fqcn is presented in the body or param or query then it determines the organization and the app. so the user should have the desired role in that org and app. but for the endpoints that fqcn is not necessary the user should have the desired role in at least one app in one org.

Reply:
ah okay got it, like the /channel/pubsub endpoint